n_beats/model_keras.py

```diff
@@ -0,0 +1,296 @@
+                plt.plot(list(range(len(x_y) - self.forecast_length, len(x_y))), validation_predictions[0, :].flatten(),
+                         color='r')
+                plt.scatter(range(len(x_y)), x_y.flatten(),
+                            color=['b'] * self.backcast_length + ['g'] * self.forecast_length)
+                plt.scatter(list(range(len(x_y) - self.forecast_length, len(x_y))),
+                            validation_predictions[0, :].flatten(), color=['r'] * self.forecast_length)
+                plt.legend(['backcast', 'forecast', 'predictions of forecast'])
+                if not os.path.exists('results/test'):
+                    os.makedirs('results/test')
+                plt.savefig('results/test/' + str(step) + '.png')
+                plt.close()
+
+        self.nbeats.save('results/n_beats_model_20_small.h5')
+
+        predictions = self.nbeats.predict(x)
+        x_y = np.concatenate([x[100, :], y[100, :]], axis=-1).flatten()
+        plt.plot(list(range(self.backcast_length)), x[100, :].flatten(), color='b')
+        plt.plot(list(range(len(x_y) - self.forecast_length, len(x_y))), y[100, :].flatten(), color='g')
+        plt.plot(list(range(len(x_y) - self.forecast_length, len(x_y))), predictions[100, :].flatten(), color='r')
+        plt.scatter(range(len(x_y)), x_y.flatten(), color=['b'] * self.backcast_length + ['g'] * self.forecast_length)
+        plt.scatter(list(range(len(x_y) - self.forecast_length, len(x_y))), predictions[100, :].flatten(),
+                    color=['r'] * self.forecast_length)
+        plt.legend(['backcast', 'forecast', 'predictions of forecast'])
+        plt.savefig('results/' + str(nb_epochs) + '.png')
+
+        validation_predictions = self.nbeats.predict(x_test)
+        x_y = np.concatenate([x_test[10, :], y_test[10, :]], axis=-1).flatten()
+        plt.figure()
+        plt.plot(list(range(self.backcast_length)), x_test[10, :].flatten(), color='b')
+        plt.plot(list(range(len(x_y) - self.forecast_length, len(x_y))), y_test[10, :].flatten(), color='g')
+        plt.plot(list(range(len(x_y) - self.forecast_length, len(x_y))), validation_predictions[10, :].flatten(),
+                 color='r')
+        plt.scatter(range(len(x_y)), x_y.flatten(), color=['b'] * self.backcast_length + ['g'] * self.forecast_length)
+        plt.scatter(list(range(len(x_y) - self.forecast_length, len(x_y))), validation_predictions[10, :].flatten(),
+                    color=['r'] * self.forecast_length)
+        plt.legend(['backcast', 'forecast', 'predictions of forecast'])
+        plt.savefig('results/test/' + str(nb_epochs) + '.png')
+
+        print(get_accuracy(y_test, validation_predictions))
+
+
+def linear_space(length, fwd_looking=True):
+    if fwd_looking:
+        t = np.linspace(0, length - 1, length)
+    else:
+        t = np.linspace(-length, -1, length)
+    t = t / length
```

Comment:
Maybe mistake here: (to be checked!)
If forecast_length=5, linear space = [0.  0.2 0.4 0.6 0.8]
and backcast_length=10, linear space = [-1.  -0.9 -0.8 -0.7 -0.6 -0.5 -0.4 -0.3 -0.2 -0.1]
I'm not sure but I think the linspace step should be the same for the backcast and the forecast.
For example [-1.  -0.9 -0.8 -0.7 -0.6 -0.5 -0.4 -0.3 -0.2 -0.1] is OK
Then forecast [0, 0.1, 0.2, 0.3, 0.4].

Reply:
Could be as simple as (again to be checked, just an idea)!

```
l = tf.linspace(-float(backcast_length), float(forecast_length),
                    backcast_length + forecast_length) / forecast_length
    l_backcast = l[:backcast_length]
    l_forecast = l[backcast_length:]
```

Reply:
That does sound better :)